lib/ws_servers/api/handlers/on_save_api_credentials.js

```diff
@@ -106,6 +106,6 @@ module.exports = async (server, ws, msg) => {
   const { userSettings } = await UserSettings.getAll()
 
   const opts = { wsURL, restURL }
-  ws.algoWorker.start({ apiKey, apiSecret, userId: DEFAULT_USER })
+  await ws.algoWorker.start({ apiKey, apiSecret, userId: DEFAULT_USER })
```

Comment:
If this fails, the host won't be available .. so, may be we should add a check in algo worker's methods to see if host is connected .. or check if the `algoworker.isStarted` is true before calling algoworker methods ? Otherwise, we will get a lot of errors `Cannot read property {method} of null`.